src/index/hnsw/faiss_hnsw.cc

```diff
@@ -1473,6 +1482,16 @@ class BaseFaissRegularIndexHNSWNode : public BaseFaissRegularIndexNode {
         const auto* data = dataset->GetTensor();
 
         const auto hnsw_cfg = static_cast<const FaissHnswConfig&>(*cfg);
+
+        const std::string metric_str = hnsw_cfg.metric_type.value();
+        if ((dim != Dim()) && (IsMetricType(metric_str, metric::COSINE) || IsMetricType(metric_str, metric::IP) ||
+                               IsMetricType(metric_str, metric::L2))) {
+            const std::string msg_e =
+                fmt::format("dimensionalities of the base dataset ({}) and the query ({}) do not match", Dim(), dim);
```

Comment:
put these repeated codes into a util function

Reply:
If I remember correctly, this should have already been intercepted in Milvus. What issue are you encountering here?